docs/adr/0008-vscode-settings.md

```diff
@@ -0,0 +1,44 @@
+# 8. Management of Visual Studio Code Settings
+
+Date: 2024-06-13
+
+## Status
+
+Accepted
+
+## Context
+
+[Settings](https://code.visualstudio.com/docs/getstarted/settings) for Visual Studio Code can be configured at a user, or workspace level.
+In both cases, they are stored in a `settings.json` file, either in an OS-dependent location in the user's home directory, or the workspace directory respectively.
+
+We used to ship a Visual Studio Code workspace settings file in the research template at `.vscode/settings.json`. Some of these pertained to GitPod (another cloud based developer machine service we used to use), and others were for user convenience.
+
+It is possible to specify user-level settings in the dev container configuration file `devcontainer.json` within the `customizations` property.
+
+[ADR 3](https://github.com/opensafely-core/research-template-docker/blob/bcecb78838dfce8c9d708af194c51e4dce541c87/docs/adr/0003-use-of-devcontainer-directory.md) states our decision to place all of our dev container related configuration within the `.devcontainer` directory.
+
+There are some settings which make the Python/ehrQL/OpenSAFELY CLI work properly in Codespaces(e.g. `python.analysis.extraPaths`, `python.terminal.activateEnvironment`, `python.defaultInterpreterPath`).
+
+There are some settings which we believe make the Codespaces UI more pleasing (e.g. `window.autDetectColorScheme`, `extensions.ignoreRecommendations`).
+
+There are some settings which we believe aid in the development workflow (e.g `files.autoSave`, `git.autofetch`)
+
+## Decision
+
+We will remove .vscode/settings.json from research template.
+
+We will add Visual Studio Code settings that we deem as useful to the research template.
+
+We will add these settings to `.devcontainer/devcontainer.json` in order to keep all codespaces/dev container config in one directory.
+
+We will remove settings that we deem as no longer needed/GitPod related.
+
+## Consequences
+
+We are still making some decisions about VS code configuration on users behalf, resulting in a user experience that might not always be to their preferences.
```

Comment:
I think we've decided that it's okay for us to make some decisions on users' behalf but not others. It would be worth teasing out what those are and being clear about them in this document. I think we've decided that we'll add VS Code config that's essential for the dev container/Codespace to function. I think we've also decided that we'll add config that's important to the user experience for all users. But importantly, we're not making decisions about things that could be considered individual preferences. Does that sound correct?